### PR TITLE
Imagenet cifar notebooks

### PR DESCRIPTION
**Pull request by vincentblot28**
# Description

- [x] I have read the [contributing guidelines](https://github.com/simai-ml/MAPIE/blob/master/CONTRIBUTING.rst)
- [x] I have updated the [HISTORY.rst](https://github.com/simai-ml/MAPIE/blob/master/HISTORY.rst) and [AUTHORS.rst](https://github.com/simai-ml/MAPIE/blob/master/AUTHORS.rst) files
- [x] Linting passes successfully : `make lint`
- [x] Typing passes successfully : `make type-check`
- [x] Unit tests pass successfully : `make tests`
- [x] Coverage is 100% : `make coverage`
- [x] Documentation builds successfully : `make doc`

### CODE REVIEW

**Comment by vtaquet** [comment]
Good job Vincent ! 
Please remove the temporary files in `.ipynb_checkpoints` and `Cifar10_files` in the `notebooks/classification` folder.

File: doc/Cifar10.rst [hunk 1]
```diff
@@ -0,0 +1,1124 @@
+Estimating prediction sets on the Cifar10 dataset
+=================================================
+
+What is done in this tutorial ?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
```
Comment (vtaquet) [inline doc/Cifar10.rst:6]: Can you start with a short introductory sentence ? Something like "In this tutorial, we will build predictions sets ..."

File: doc/Cifar10.rst [hunk 2]
```diff
@@ -0,0 +1,1124 @@
...
+   -  **Cifar10 dataset** : 10 classes (horse, dog, cat, frog, deer,
+      bird, airplane, truck, ship, automobile)
+
+..
+
+   -  Use ``mapie.classification.MapieClassifier`` to compare the
+      prediction sets estimated by several conformal methods on the
+      Cifar10 dataset.
+
+   -  Train a small CNN to predict the image classe
```
Comment (vtaquet) [inline doc/Cifar10.rst:16]: Typo "class"

File: doc/Cifar10.rst [hunk 3]
```diff
@@ -0,0 +1,1124 @@
...
+
+..
+
+   -  Create a custom classe ``TensorflowToMapie`` to resolve adherence
```
Comment (vtaquet) [inline doc/Cifar10.rst:20]: Typo "class"

File: doc/Cifar10.rst [hunk 4]
```diff
@@ -0,0 +1,1124 @@
...
+    SPACE_IN_SUBPLOTS = 4.0
+    FONT_SIZE = 18
+
+
+1. Data loading
+---------------
+
+The Cifar10 dataset is downloaded from the ``Tensorflow Datasets``
+library. The training set is then splitted into a training, validation
+and a calibration set which will be used as follow: > - **Training
```
Comment (vtaquet) [inline doc/Cifar10.rst:68]: Add a double backspace to allow the bullet points to rend in the documentation.

File: doc/Cifar10.rst [hunk 5]
```diff
@@ -0,0 +1,1124 @@
...
+    def train_valid_calib_split(
+        X: np.ndarray,
+        y: np.ndarray,
+        calib_size: float = .1,
+        val_size: float = .33,
+        random_state: int = 42
+    
+    ) -> Tuple[np.ndarray, np.ndarray, np.ndarray, np.ndarray, np.ndarray, np.ndarray]:
+        """
+        Create calib and valiv dataset from the train dataset.
```
[the remark this answers is not in the document]
Comment (vtaquet) [reply doc/Cifar10.rst:85]: datasetS

File: doc/Cifar10.rst [hunk 6]
```diff
@@ -0,0 +1,1124 @@
...
+            Dense(32, activation='relu'),
+            Dense(10, activation='softmax'),
+        ])
+        model.compile(loss=loss, optimizer=optimizer, metrics=metrics)
+        return model
+
+Training the algorithm with a custom class called ``TensorflowToMapie``
+-----------------------------------------------------------------------
+
+As MAPIE asked that the model has a ``fit``, ``predict_proab``,
```
Comment (vtaquet) [inline doc/Cifar10.rst:322]: Typo "predict_proba"

File: doc/Cifar10.rst [hunk 7]
```diff
@@ -0,0 +1,1124 @@
...
+            Tuple indicating the indices of the image to put
+            the text on.
+        
+        position: float
+            Position of the text on the image.
+        
+        label_name: str
+            Name of the label to plot.
+        
+        proab: float
```
Comment (vtaquet) [inline doc/Cifar10.rst:757]: Typo "proab"

File: doc/Cifar10.rst [hunk 8]
```diff
@@ -0,0 +1,1124 @@
...
+
+What is done in this tutorial ?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+   -  **Cifar10 dataset** : 10 classes (horse, dog, cat, frog, deer,
+      bird, airplane, truck, ship, automobile)
+
+..
+
+   -  Use ``mapie.classification.MapieClassifier`` to compare the
```
Comment (vtaquet) [inline doc/Cifar10.rst:12]: :class:`mapie.classification.MapieClassifier` to create a link towards the API

File: doc/Cifar10.rst [hunk 9]
```diff
@@ -0,0 +1,1124 @@
...
+1. Data loading
+---------------
+
+The Cifar10 dataset is downloaded from the ``Tensorflow Datasets``
+library. The training set is then splitted into a training, validation
+and a calibration set which will be used as follow: > - **Training
+set**: used to train our neural network. > - **Validation set**: used to
+check that our model is not overfitting. > - **Calibration set**: used
+to calibrate the conformal scores in
+``mapie.classification.MapieClassifier``
```
Comment (vtaquet) [inline doc/Cifar10.rst:72]: :class:`mapie.classification.MapieClassifier`

File: doc/Cifar10.rst [hunk 10]
```diff
@@ -0,0 +1,1124 @@
...
+
+    y_true = label_binarize(y=y_test, classes=np.arange(max(y_test)+1))
+    y_pred_proba = cirfar10_model.predict_proba(X_test)
+    y_pred = cirfar10_model.predict(X_test)
+
+
+3. Prediction of the prediction sets
+------------------------------------
+
+We will now estimate the prediction sets with the five conformal methods
```
Comment (vtaquet) [inline doc/Cifar10.rst:495]: :class:`mapie.classification.MapieClassifier`

File: doc/Cifar10.rst [hunk 11]
```diff
@@ -0,0 +1,1124 @@
...
+methods give a marginal coverage close to the desired one, i.e. 90%.
+Notice that the “cumulated_score” method, which always includes the last
+label whose cumulated score is above the given quantile, tends to give
+slightly higher marginal coverages since the prediction sets are
+slightly too big.
+
+4. Visualization of the prediction sets
+---------------------------------------
+
+Thanks to this really awesome function, you can generate an amazing
```
Comment (vtaquet) [inline doc/Cifar10.rst:667]: Ahah, this sentence should be removed now :-)

File: doc/Cifar10.rst [hunk 12]
```diff
@@ -0,0 +1,1124 @@
...
+    plt.axhline(0.9, ls="--", color="k")
+    plt.ylabel("Conditional coverage")
+    plt.legend(loc=[1, 0])
+
+
+
+
+.. parsed-literal::
+
+    <matplotlib.legend.Legend at 0x16a0602b0>
```
Comment (vtaquet) [inline doc/Cifar10.rst:981]: To remove ?

File: doc/Cifar10.rst (hunk 2 above)
Comment (vincentblot28) [reply doc/Cifar10.rst:16]: done

File: doc/Cifar10.rst (hunk 3 above)
Comment (vincentblot28) [reply doc/Cifar10.rst:20]: done

File: doc/Cifar10.rst (hunk 5 above)
[the remark this answers is not in the document]
Comment (vincentblot28) [reply doc/Cifar10.rst:85]: done

File: doc/Cifar10.rst (hunk 6 above)
Comment (vincentblot28) [reply doc/Cifar10.rst:322]: done

File: doc/Cifar10.rst (hunk 7 above)
Comment (vincentblot28) [reply doc/Cifar10.rst:757]: done

File: doc/Cifar10.rst [hunk 13]
```diff
@@ -0,0 +1,1124 @@
...
+        def __sklearn_is_fitted__(self):
+            if self.trained_:
+                return True
+            else:
+                return False
+
+.. code-block:: python
+
+    model = get_model(
+                    input_shape=(32, 32, 3),
```
[the remark this answers is not in the document]
Comment (vincentblot28) [reply doc/Cifar10.rst:429]: done

File: doc/Cifar10.rst (hunk 9 above)
Comment (vincentblot28) [reply doc/Cifar10.rst:72]: done

File: doc/Cifar10.rst (hunk 10 above)
Comment (vincentblot28) [reply doc/Cifar10.rst:495]: done

File: doc/Cifar10.rst (hunk 11 above)
Comment (vincentblot28) [reply doc/Cifar10.rst:667]: done

File: doc/Cifar10.rst (hunk 12 above)
Comment (vincentblot28) [reply doc/Cifar10.rst:981]: yes ! done

File: doc/Cifar10.rst [hunk 14]
```diff
@@ -0,0 +1,1124 @@
...
+.. parsed-literal::
+
+    <matplotlib.legend.Legend at 0x16a0602b0>
+
+
+
+
+.. parsed-literal::
+
+    <Figure size 432x288 with 0 Axes>
```
[the remark this answers is not in the document]
Comment (vincentblot28) [reply doc/Cifar10.rst:988]: Yes ! done

File: doc/Cifar10.rst [hunk 15]
```diff
@@ -0,0 +1,1108 @@
...
+            Flatten(),
+            Dense(128, activation='relu'),
+            Dense(64, activation='relu'),
+            Dense(32, activation='relu'),
+            Dense(10, activation='softmax'),
+        ])
+        model.compile(loss=loss, optimizer=optimizer, metrics=metrics)
+        return model
+
+Training the algorithm with a custom class called ``TensorflowToMapie``
```
Comment (vtaquet) [inline doc/Cifar10.rst:321]: Section without number.

File: doc/Cifar10.rst [hunk 16]
```diff
@@ -0,0 +1,1108 @@
...
+
+What is done in this tutorial ?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+   -  **Cifar10 dataset** : 10 classes (horse, dog, cat, frog, deer,
+      bird, airplane, truck, ship, automobile)
+
+..
+
+   -  Use :class:``mapie.classification.MapieClassifier`` to compare the
```
Comment (vtaquet) [inline doc/Cifar10.rst:12]: Only one ` is needed before and after mapie.classification.MapieClassifier.

File: doc/Cifar10.rst (hunk 8 above)
Comment (vtaquet) [reply doc/Cifar10.rst]: Single "`"

File: doc/Cifar10.rst [hunk 17]
```diff
@@ -0,0 +1,1108 @@
...
+
+The Cifar10 dataset is downloaded from the ``Tensorflow Datasets``
+library. The training set is then splitted into a training, validation
+and a calibration set which will be used as follow:
+
+   -  **Training set**: used to train our neural network.
+   -  **Validation set**: used to check that our model is not
+      overfitting.
+   -  **Calibration set**: used to calibrate the conformal scores in
+      :class:``mapie.classification.MapieClassifier``
```
[the remark this answers is not in the document]
Comment (vincentblot28) [reply doc/Cifar10.rst]: done

File: doc/Cifar10.rst (hunk 16 above)
Comment (vincentblot28) [reply doc/Cifar10.rst]: done

File: doc/Cifar10.rst (hunk 4 above)
Comment (vincentblot28) [reply doc/Cifar10.rst]: done

File: doc/Cifar10.rst (hunk 15 above)
Comment (vincentblot28) [reply doc/Cifar10.rst]: done

File: doc/Cifar10.rst (hunk 1 above)
Comment (vincentblot28) [reply doc/Cifar10.rst:11]: done

**Pull request merged**